crates/node/core/src/args/network.rs

```diff
@@ -148,9 +148,29 @@ pub struct NetworkArgs {
     /// Max capacity of cache of hashes for transactions pending fetch.
     #[arg(long = "max-tx-pending-fetch", value_name = "COUNT", default_value_t = DEFAULT_MAX_CAPACITY_CACHE_PENDING_FETCH, verbatim_doc_comment)]
     pub max_capacity_cache_txns_pending_fetch: u32,
+
+    /// Name of network interface used to communicate with peers.
+    ///
+    /// If flags is set, but no value is passed, the default interface for docker `eth0` is tried.
+    #[cfg(not(target_os = "windows"))]
+    #[arg(long = "net-if", conflicts_with = "addr", value_name = "IF_NAME")]
+    pub net_if: Option<String>,
 }
 
 impl NetworkArgs {
+    /// Returns the resolved IP address.
+    pub fn resolved_addr(&self) -> eyre::Result<IpAddr> {
```

Comment:
could we make this infallible?

unwrap_or_else(self.addr)

then we don't need to return a result for fn network_config for a feature that isn't often used and likely doesn't fail and has a fallback

Reply:
the arg `net-if` conflicts with `addr`, can return the default address